include/polycap.h

```diff
@@ -108,7 +108,7 @@ void polycap_rng_free(polycap_rng *rng);
 // exposing more rng functions may be useful..
 
 // construct a new random polycap_photon 
-polycap_photon* polycap_source_get_photon(polycap_source *source, polycap_description *description, polycap_rng *rng, size_t n_energies, double *energies);
+polycap_photon* polycap_source_get_photon(polycap_source *source, polycap_description *description, polycap_rng *rng, size_t n_energies, double *energies, polycap_vector3 *src_start_coords);
```

Comment:
Dit kan properder gedaan worden door de volgende functie te implementeren:

```c
polycap_vector3 polycap_photon_get_start_coords(polycap_photon *photon);
```

Reply:
Nee, is niet helemaal hetzelfde.
`photon.start_coords` zijn de coordinaten van het foton bij begin van simulatie, aan de ingang van het polycapillair (dus wat jij in msim zou meegeven om dan naar photon_launch te gaan).
Wat ik hier wil verkrijgen is het coordinaat van mijn gesimuleerde source van waar de X-straal komt, geeft me dus uiteindelijk het profiel van de bron (waardoor ik al wat bugjes heb gezien die momenteel nog in de code zitten, mijn bron wordt immers gesimuleerd met een gat in het midden heb ik de indruk)

Reply:
azo... Is dit dan enkel iets voor debuggen? Indien zo, zou ik in de definitieve API wel dat argument terug uit de functie verwijderen...

Reply:
Goh, daar ben ik nog niet uit.
Aan de ene kant is het wel leuk om de gesimuleerde bron coordinaten mee te geven, kan de gebruiker nuttigen om exact te zien hoe de bron eruit zag.
Aan de andere kant is het natuurlijk de gebruiker die de bron parameters meegeeft, dus op zich zou er weinig verrassends aan het profiel mogen zijn :P 

An sich zie ik het niet zo als een probleem voor gebruikers van de API. Bvb in msim zal de `polycap_photon_get_start_coords` functie wss niet gebruikt worden (behalve mss voor de focussing kant van de zaken), en indien het wel gebruikt wordt kan je in principe gewoon een dummy `polycap_vector3` meegeven, wat dus nauwelijks geheugen vergt.

Reply:
Ok vergeet dan maar die `polycap_photon_get_start_coords` 😄. 

In de regel gebruikt ge wel altijd best getters en setters voor het foefelen met de member variables van de structs. Alhoewel C u de mogelijkheid geeft variabelen te returnen via de argument list, is dat toch zelden de beste oplossing, en kan verwarrend zijn voor minder ervaren programmeurs. Dit is trouwens iets dat in de meeste talen niet eens mogelijk is (Python!)